src/vm/corbundle.cpp

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//*****************************************************************************
+// CorHost.cpp
+//
+// Implementation for the meta data dispenser code.
+//
+
+//*****************************************************************************
+
+#include "common.h"
+#include "corbundle.h"
+#include <utilcode.h>
+#include <corhost.h>
+
+static LPCSTR UnicodeToUtf8(LPCWSTR str)
+{
+    STANDARD_VM_CONTRACT;
+
+    int length = WideCharToMultiByte(CP_UTF8, 0, str, -1, NULL, 0, 0, 0);
+    _ASSERTE(length != 0);
+
+    LPSTR result = new (nothrow) CHAR[length];
+    _ASSERTE(result != NULL);
+
+    length = WideCharToMultiByte(CP_UTF8, 0, str, -1, result, length, 0, 0);
+    _ASSERTE(length != 0);
+
+    return result;
+}
+
+BundleInfo::BundleInfo(LPCWSTR bundlePath, bool(*probe)(LPCSTR, INT64*, INT64*))
+{
+    STANDARD_VM_CONTRACT;
+
+    m_path = bundlePath;
+    m_probe = probe;
+
+    // In this prototype, the bundle-base path is simply the directory containing 
+    // the single-file bundle. When the Probe() function searches within the bundle,
+    // it masks out the base_path from the assembly-path (if found).
+    //
+    //  For example: 
+    //  BundleInfo.Probe("lib.dll") => m_probe("lib.dll")
+    //  BundleInfo.Probe("path/to/exe/lib.dll") => m_probe("lib.dll")
+    //  BundleInfo.Probe("path/to/exe/and/some/more/lib.dll") => m_probe("and/some/more/lib.dll")
+    //
+    // This strategy obviously hides any actual files in path/to/exe from being loaded.
```

Comment:
For the prototype, I think we should not have the same assembly in the two locations: bundle and next to the single-file. 

If we want to process some assemblies from actual files (for debugging), it is better to exclude them from the bundle (especially since bundle construction is manual in the prototype).